sensor/kubernetes/listener/resources/registrymirrorsets_test.go

```diff
@@ -28,60 +29,92 @@ func TestProcessEvent(t *testing.T) {
 	}
 	for _, tc := range tt {
 		t.Run(tc.action.String(), func(t *testing.T) {
-			rs := registry.NewRegistryStore(nil)
-			d := newRegistryMirrorDispatcher(rs)
+			ms := &FakeMirrorStore{}
+			d := newRegistryMirrorDispatcher(ms)
+
 			d.ProcessEvent(icspA, nil, tc.action)
+			assert.True(t, ms.upsertICSPInvoked)
+			assert.False(t, ms.deleteICSPInvoked)
+
 			d.ProcessEvent(idmsA, nil, tc.action)
+			assert.True(t, ms.upsertIDMSInvoked)
+			assert.False(t, ms.deleteIDMSInvoked)
+
 			d.ProcessEvent(itmsA, nil, tc.action)
-			icspRules, idmsRules, itmsRules := rs.GetAllMirrorSets()
-			assert.Len(t, icspRules, 1)
-			assert.Len(t, idmsRules, 1)
-			assert.Len(t, itmsRules, 1)
+			assert.True(t, ms.upsertITMSInvoked)
+			assert.False(t, ms.deleteITMSInvoked)
 		})
 	}
 
-	t.Run("no panic when removing non existant resource", func(t *testing.T) {
-		rs := registry.NewRegistryStore(nil)
-		d := newRegistryMirrorDispatcher(rs)
+	t.Run("no panic when removing non existent resource", func(t *testing.T) {
+		ms := &FakeMirrorStore{}
+		d := newRegistryMirrorDispatcher(ms)
 		action := central.ResourceAction_REMOVE_RESOURCE
 		d.ProcessEvent(icspA, nil, action)
-		d.ProcessEvent(idmsA, nil, action)
-		d.ProcessEvent(itmsA, nil, action)
-	})
+		assert.False(t, ms.upsertICSPInvoked)
+		assert.True(t, ms.deleteICSPInvoked)
 
-	t.Run("upsert followed by delete removes appropriate resource", func(t *testing.T) {
-		rs := registry.NewRegistryStore(nil)
-		d := newRegistryMirrorDispatcher(rs)
-		action := central.ResourceAction_CREATE_RESOURCE
-		d.ProcessEvent(icspA, nil, action)
 		d.ProcessEvent(idmsA, nil, action)
-		d.ProcessEvent(itmsA, nil, action)
-
-		action = central.ResourceAction_REMOVE_RESOURCE
-		d.ProcessEvent(icspA, nil, action)
-		icspRules, idmsRules, itmsRules := rs.GetAllMirrorSets()
-		assert.Len(t, icspRules, 0)
-		assert.Len(t, idmsRules, 1)
-		assert.Len(t, itmsRules, 1)
-
-		d.ProcessEvent(idmsA, nil, action)
-		icspRules, idmsRules, itmsRules = rs.GetAllMirrorSets()
-		assert.Len(t, icspRules, 0)
-		assert.Len(t, idmsRules, 1)
-		assert.Len(t, itmsRules, 0)
+		assert.False(t, ms.upsertIDMSInvoked)
+		assert.True(t, ms.deleteIDMSInvoked)
 
 		d.ProcessEvent(itmsA, nil, action)
-		icspRules, idmsRules, itmsRules = rs.GetAllMirrorSets()
-		assert.Len(t, icspRules, 0)
-		assert.Len(t, idmsRules, 0)
-		assert.Len(t, itmsRules, 0)
+		assert.False(t, ms.upsertITMSInvoked)
+		assert.True(t, ms.deleteITMSInvoked)
 	})
 
 	t.Run("unknown type panics for non-release builds", func(t *testing.T) {
 		if !buildinfo.ReleaseBuild {
-			rs := registry.NewRegistryStore(nil)
-			d := newRegistryMirrorDispatcher(rs)
+			ms := &FakeMirrorStore{}
+			d := newRegistryMirrorDispatcher(ms)
 			assert.Panics(t, func() { d.ProcessEvent(nil, nil, 0) })
 		}
 	})
 }
+
+type FakeMirrorStore struct {
```

Comment:
This also doesn't need to be exported.

On the other hand, you could also choose to generate a mock for the mirror store via `//go:generate mockgen-wrapper` instead of writing the mock by yourself.

Reply:
I'll probably go the mock route - I noticed running a unit test takes 5+ seconds to run when using the generated mocks, whereas when using a type like this is instant.

Reply:
![image](https://github.com/stackrox/stackrox/assets/119438707/57b0e9e6-cedd-4ebc-82c4-29a795307c02)